src/views/Dashboard/components/Widgets/TransactionsWidget/TransactionsWidget.tsx

```diff
@@ -89,155 +44,38 @@ const headCells: readonly IHeadCellProps[] = [
-  order: Order;
-  orderBy: string;
-  rowCount: number;
-}
-
-function EnhancedTableHead(props: EnhancedTableProps) {
-  const { order, orderBy, onRequestSort } = props;
-  const createSortHandler = (property: keyof Data) => (event: React.MouseEvent<unknown>) => {
-    onRequestSort(event, property);
-  };
-
-  return (
-    <TableHead>
-      <TableRow>
-        {headCells.map((headCell) => (
-          <TableCell
-            key={headCell.id}
-            align={headCell.numeric ? "right" : "left"}
-            padding={headCell.disablePadding ? "none" : "normal"}
-            sortDirection={orderBy === headCell.id ? order : false}
-          >
-            <TableSortLabel
-              active={orderBy === headCell.id}
-              direction={orderBy === headCell.id ? order : "asc"}
-              onClick={createSortHandler(headCell.id)}
-            >
-              {headCell.label}
-              {orderBy === headCell.id ? (
-                <Box component="span" sx={visuallyHidden}>
-                  {order === "desc" ? "sorted descending" : "sorted ascending"}
-                </Box>
-              ) : null}
-            </TableSortLabel>
-          </TableCell>
-        ))}
-      </TableRow>
-    </TableHead>
-  );
-}
-
-const EnhancedTableToolbar = () => {
-  return (
-    <Toolbar>
-      {
-        <Typography sx={{ flex: "1 1 100%" }} variant="h6" id="tableTitle">
-          <SLink href="/transactions">My Transactions</SLink>
-        </Typography>
-      }
-      {/* {
-        <Tooltip title="Filter list">
-          <IconButton>
-            <FilterListIcon />
-          </IconButton>
-        </Tooltip>
-      } */}
-    </Toolbar>
-  );
-};
-
 const TransactionsWidget: React.FC = () => {
-  const [order, setOrder] = React.useState<Order>("asc");
-  const [orderBy, setOrderBy] = React.useState<keyof Data>("date");
-  const [page, setPage] = React.useState(0);
-  const [rowsPerPage, setRowsPerPage] = React.useState(3);
   const { transactions } = useMyTxs();
 
-  let emptyRows;
+  let txRows;
 
-  const handleRequestSort = (event: React.MouseEvent<unknown>, property: keyof Data) => {
-    const isAsc = orderBy === property && order === "asc";
-    setOrder(isAsc ? "desc" : "asc");
-    setOrderBy(property);
-  };
-
-  const handleChangePage = (event: unknown, newPage: number) => {
-    setPage(newPage);
-  };
-
-  const handleChangeRowsPerPage = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setRowsPerPage(parseInt(event.target.value, 10));
-    setPage(0);
-  };
-
-  if (transactions !== undefined) {
-    // Avoid a layout jump when reaching the last page with empty rows.
-    emptyRows = page > 0 ? Math.max(0, (1 + page) * rowsPerPage - transactions.length) : 0;
-  } else {
-    emptyRows = 0;
+  if (transactions) {
+    txRows = transactions.map((row, index) => {
+      return (
+        <TableRow hover tabIndex={-1} key={row.timestamp + "-" + index}>
```

Comment:
when making the table component generic i wasn't expecting components like this to be created here and passed in. while it's flexible, and may ultimately be what you want, consider the layers of interaction here.

what i was expecting:
1. gather table data into an array (map), adding extra properties for text alignment, etc, as may be wanted
2. send that data into the table component to display it (along with the header data, footer data, etc)

the reason I was expecting this is because the JupTable component should be in charge of creating the components to display the data, but the component that needs a table will know what data should be displayed. they both have their own responsibilities. if you keep the tableRow/tableCell outside of the JupTable component, what happens someday when you have to change how the table component is rendered, and each of the places you use it needs to be heavily updated. whereas if each use-case only gathered data into an array, then the refactoring would likely be much less, plus there would be fewer import statements.

obviously this is a suggestion, it does work as-is, but i can see it improving quality to make this change sooner than later.

Reply:
I made an issue on this: https://github.com/Up-North-Coding/jovian/issues/34